File: examples/reduce.f90
PROGRAM reduce

USE mpi_f08
IMPLICIT NONE

INTEGER myid, nproc, root
REAL A(2), res(2)

    CALL MPI_Init()
    CALL MPI_Comm_size(MPI_COMM_WORLD, nproc)
    CALL MPI_Comm_rank(MPI_COMM_WORLD, myid)

    root = 0
    a(1) = 2 * myid + 0
    a(2) = 2 * myid + 1

    CALL MPI_REDUCE(a, res, 2, MPI_REAL, MPI_SUM, root, &
                    MPI_COMM_WORLD)
    IF( myid == root ) THEN
        WRITE(*,*) myid, ': res(1)=', res(1), &
                           'res(2)=', res(2)
    END IF

    CALL MPI_Finalize()

END PROGRAM reduce

File: exemples/reduce.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char * argv[]) {
    int rank, size, root = 0;
    float a[2], res[2];

    MPI_Init( &argc, &argv );
    MPI_Comm_rank( MPI_COMM_WORLD, &rank );
    MPI_Comm_size( MPI_COMM_WORLD, &size );

    a[0] = 2 * rank + 0;
    a[1] = 2 * rank + 1;

    MPI_Reduce( a, res, 2, MPI_FLOAT, MPI_SUM, root, MPI_COMM_WORLD );

    if (rank == root) {
        printf( "%d: res[0]=%f\tres[1]=%f\n", root, res[0], res[1] );
    }

    MPI_Finalize();
    return 0;
}

